esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -2016,18 +2013,18 @@ void WsWuInfo::getWorkunitThorSlaveLog(IGroup *nodeGroup, const char *ipAddress,
     dir.clear();
     // logfile is now the file to load
 
-    OwnedIFileIO rIO = logfile->openShared(IFOread,IFSHfull);
-    if (!rIO)
-        throw MakeStringException(ECLWATCH_CANNOT_READ_FILE,"Cannot read file %s.",logName.str());
-
-    OwnedIFileIOStream ios = createIOStream(rIO);
     if (slaveNum > 0)
     {
+        OwnedIFileIO rIO = logfile->openShared(IFOread,IFSHfull);
+        if (!rIO)
+            throw MakeStringException(ECLWATCH_CANNOT_READ_FILE,"Cannot read file %s.",logName.str());
+
+        OwnedIFileIOStream ios = createIOStream(rIO);
         readWorkunitLog(ios, buf, outFile);
     }
     else
     {//legacy wuid
-        appendIOStreamContent(buf, ios, forDownload);
+        readFileContent(logfile, logName, buf, forDownload, nullptr);
```

Comment:
There's only 3 places readFileContent is called and this is the only one that reads into a buffer,
It would be clearer if the method wasn't combined to take both a IFileIO and a MemoryBuffer and test which is passed,

i.e. it would clearer in the 1 case (here) where's reading into MemoryBuffer to put that code into it's own method and call it here - if you think it is likely to be reused (but perhaps not as legacy), or move that code here.

readFileContent becomes clean then, handling copying a file and error checking (see other comments).

Reply:
Will move the copy file part to new copyRemoteFileContent().